onyo/lib/onyo.py

```diff
@@ -203,155 +203,72 @@ def clear_caches(self,
+            A format string defining the commit message subject to generate.
 
         max_length: int
-            Limit the length of the generated message to `max_length` if
-            possible.
+            An integer specifying the maximal length for generated commit message subjects.
+
+        **kwargs
+            Values to insert into the `format_string`. If values are paths, they will be shortened
+            to include as much user readable information as possible.
 
         Returns
         -------
         str
-            A commit message subject to use with `git commit`.
+            A message suitable as a commit message subject for use with `git commit`.
         """
-
         # long message: full paths (relative to repo-root)
-        paths_str = ','.join([f"'{x}'" for x in changes])
-        msg = f"{msg_dummy}: {paths_str}"
-        if destination:
-            msg = f"{msg} -> '{destination}'"
-
-        if len(msg) < max_length:
-            return msg
+        shortened_kwargs = {}
+        for elem in kwargs:
+            if isinstance(kwargs[elem], list):
+                shortened_kwargs[elem] = ','.join([f"'{x}'" for x in kwargs[elem]])
+            else:
+                shortened_kwargs[elem] = str(kwargs[elem])
+        message = format_string.format(**shortened_kwargs)
+        if len(message) < max_length:
+            return message
 
         # medium message: highest level (e.g. dir or asset name)
-        paths = [x.name for x in changes]
-        paths_str = ','.join(["'{}'".format(x) for x in paths])
-        msg = f"{msg_dummy}: {paths_str}"
-        if destination:
-            msg = f"{msg} -> '{destination.relative_to(destination.parent)}'"
-
-        if len(msg) < max_length:
-            return msg
+        shortened_kwargs = {}
+        for elem in kwargs:
+            if isinstance(kwargs[elem], list):
+                shortened_kwargs[elem] = ','.join([f"'{x.name}'" if isinstance(x, Path) else
+                                                   f"'{x}'" for x in kwargs[elem]])
+            elif isinstance(kwargs[elem], Path):
+                shortened_kwargs[elem] = str(kwargs[elem].name)
+            else:
+                shortened_kwargs[elem] = str(kwargs[elem])
+        message = format_string.format(**shortened_kwargs)
+        if len(message) < max_length:
+            return message
 
         # short message: "type" of devices in summary (e.g.  "laptop (2)")
-        paths = [x.name.split('_')[0] for x in changes]
-        paths_str = ','.join(sorted(["'{} ({})'".format(x, paths.count(x))
-                                     for x in set(paths)]))
-        msg = f"{msg_dummy}: {paths_str}"
-        if destination:
-            msg = f"{msg} -> '{destination.relative_to(destination.parent)}'"
+
+        shortened_kwargs = {}
+        for elem in kwargs:
+            if isinstance(kwargs[elem], list):
+                paths = [x.name.split('_')[0] for x in kwargs[elem]]
```

Comment:
As pointed out in #462, the file name splitting by `_` as well as assuming the first element of such a split being a meaningful grouping for a shorter depiction is not valid anymore.
Entirely depends on the `onyo.assets.filename` config.

I think, that entire block can be deleted.

Reply:
You are right, but I wonder if there is not a reasonable replacement for this; e.g. the first element of `onyo.assets.filename` and/or the biggest overlap in the name of all  things the list or something like that.

Reply:
Any such approach is making assumptions about what kind of configs would be used. We have no real-world evidence for any of that.
Ultimately, this is collapsing assets into some meaningful grouping. We have no way of assessing what would be meaningful in general, though. So, I think the only reasonable way to do that, is to have a configuration that specifies what to use in this fashion.